catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/action/SourceActionProviderImpl.java

```diff
@@ -0,0 +1,119 @@
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.impl.action;
+
+import ddf.action.Action;
+import ddf.action.ActionProvider;
+import ddf.action.impl.ActionImpl;
+import ddf.catalog.source.Source;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.boon.json.JsonFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Convert a JSON string to a map of source identifiers to action URLs. */
+public class SourceActionProviderImpl implements ActionProvider {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SourceActionProviderImpl.class);
+
+  private static final String TITLE = "title";
+
+  private static final String URL = "url";
+
+  private Map<String, Action> sourceIdToAction;
+
+  private String actionProviderId;
+
+  public SourceActionProviderImpl(String actionProviderId) {
+    this.actionProviderId = actionProviderId;
+  }
+
+  @Override
+  public synchronized <T> Action getAction(T subject) {
+
+    if (!(subject instanceof Source)) {
+      return null;
+    }
+
+    Source source = (Source) subject;
+
+    return sourceIdToAction.get(source.getId());
+  }
+
+  @Override
+  public String getId() {
+    return actionProviderId;
+  }
+
+  /**
+   *
+   *
+   * <pre>
+   *     {
+   *         "sourceId1": {
+   *             "title": "Title",
+   *             "url": "http://example.com"
+   *         }
+   *     }
+   * </pre>
+   *
+   * @param sourceActionsJson json representing the source actions
+   */
+  public synchronized void setSourceActions(String sourceActionsJson) {
+    if (StringUtils.isEmpty(sourceActionsJson)) {
+      this.sourceIdToAction = Collections.emptyMap();
+    } else {
+      Map<String, Map<String, String>> rawMap =
+          JsonFactory.create().readValue(sourceActionsJson, Map.class);
+      if (rawMap != null) {
+
+        this.sourceIdToAction =
+            rawMap
+                .entrySet()
+                .stream()
+                .map(entry -> new ImmutablePair<>(entry.getKey(), toAction(entry.getValue())))
+                .filter(stringActionPair -> Objects.nonNull(stringActionPair.getValue()))
+                .collect(Collectors.toMap(Pair::getKey, Pair::getValue));
+
+      } else {
+        this.sourceIdToAction = Collections.emptyMap();
+      }
+    }
+  }
+
+  private Action toAction(Map<String, String> mapOfStrings) {
+    if (mapOfStrings.containsKey(TITLE) && mapOfStrings.containsKey(URL)) {
+      try {
+        return new ActionImpl(getId(), mapOfStrings.get(TITLE), "", new URL(mapOfStrings.get(URL)));
```

Comment:
No description passed into the ActionImpl constructor?

Reply:
I can add a description field to the JSON configured in the admin ui, but i'll need to figure out a good way to display that description to the user in the ui

Reply:
There are other descriptions we have but do not display in Intrigue.  It is ok to have the metadata.  It is not required that all UIs display it.

Reply:
I've added a description field to the config and the text appears under the 'help' feature in the UI